src/pages/index.js

```diff
@@ -1,15 +1,31 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import ProductList from '../components/ProductList';
-import { useEffect, useState } from 'react';
-import axios from 'axios';
+import { fetchProducts } from '../store/productSlice';
+import { ToastContainer, toast } from 'react-toastify'; // Import ToastContainer and toast
+import 'react-toastify/dist/ReactToastify.css'; // Import Toastify styles
 
 export default function HomePage() {
-  const [products, setProducts] = useState([]);
+const dispatch = useDispatch();
+const products = useSelector(state => state.products.products);
+const status = useSelector(state => state.products.status);
+const error = useSelector(state => state.products.error);
```

Comment:
no need to store error as toast already shows , error is un-necessary to keep in redux state